tests/test_omegaconf.py

```diff
@@ -518,3 +518,93 @@ def test_resolve(cfg: Any, expected: Any) -> None:
 def test_resolve_invalid_input() -> None:
     with raises(ValueError):
         OmegaConf.resolve("aaa")  # type: ignore
+
+
+@mark.parametrize(
+    ("cfg_params,clear_resolver_params,expected"),
+    [
+        # tuple
+        (
+            # cfg_params
+            dict(
+                name="TR1",
+                resolver=lambda x: str(x).lower(),
+                use_cache=False,
+                replace=True,
+            ),
+            # clear_resolver_params
+            dict(name="TR1"),
+            # expected
+            dict(pre_addition=False, post_addition=True, post_removal=False),
+        ),
+        # tuple
+        (
+            # cfg_params
+            dict(
+                name="TR2",
+                resolver=lambda x: str(x).upper(),
+                use_cache=False,
+                replace=True,
+            ),
+            # clear_resolver_params
+            dict(name="TR2"),
+            # expected
+            dict(pre_addition=False, post_addition=True, post_removal=False),
+        ),
+        # tuple
+        (
+            # cfg_params
+            dict(
+                name="TR3",
+                resolver=lambda x: str(x).upper(),
+                use_cache=False,
+                replace=False,
+            ),
+            # clear_resolver_params
+            dict(name="TR3"),
+            # expected
+            dict(pre_addition=False, post_addition=True, post_removal=False),
+        ),
```

Comment:
I don't think we need to test all of these combinations of `use_cache`/`resolver`/`replace` -- we are trying to test `OmegaConf.clear_resolver`, not `OmegaConf.register_resolver`. It should be good enough to test these cases:
  - a custom resolver (registered by the user)
  - a default resolver (e.g. `oc.env`)
  - a non-existing resolver

Reply:
Ok. I will slim it down. Thank you.

Reply:
Done!